packages/ckeditor5-clipboard/src/clipboardmarkersutils.ts

```diff
@@ -262,76 +385,121 @@ export default class ClipboardMarkersUtils extends Plugin {
-
-		return Object
-			.entries( fakeFragmentMarkersInMap )
-			.reduce<Record<string, Range>>( ( acc, [ markerName, [ startElement, endElement ] ] ) => {
-				// Marker is not entire inside specified fragment but rather starts or ends inside it.
-				if ( !endElement ) {
-					const type = startElement.getAttribute( 'data-type' );
-
-					if ( type === 'end' ) {
-						// <fake-marker> [ phrase</fake-marker> phrase ]
-						//   ^
-						// Handle case when marker is just before start of selection.
-
-						const endPosition = writer.createPositionAt( startElement, 'before' );
-						const startPosition = writer.createPositionFromPath( endPosition.root, [ 0 ] );
-
-						writer.remove( startElement );
-						acc[ markerName ] = new Range( startPosition, endPosition );
-					} else {
-						// [<fake-marker>phrase]</fake-marker>
-						//                           ^
-						// Handle case when fake marker is after selection.
-
-						const startPosition = writer.createPositionAt( startElement, 'before' );
-						writer.remove( startElement );
-
-						const endPosition = writer.createPositionAt( rootElement, 'end' );
-						acc[ markerName ] = new Range( startPosition, endPosition );
-					}
-
-					return acc;
-				}
+		const fakeMarkersElements = this._getAllFakeMarkersFromElement( writer, rootElement );
+		const fakeMarkersRanges = fakeMarkersElements.reduce<Record<string, FakeMarkerRangeConstruct>>( ( acc, fakeMarker ) => {
+			const position = fakeMarker.markerElement && writer.createPositionBefore( fakeMarker.markerElement );
+			let prevFakeMarker: FakeMarkerRangeConstruct | null = acc[ fakeMarker.name ];
+
+			// Handle scenario when tables clone cells with the same fake node. Example:
+			//
+			// <cell><fake-marker-a></cell> <cell><fake-marker-a></cell> <cell><fake-marker-a></cell>
+			//                                          ^ cloned                    ^ cloned
+			//
+			// The easiest way to bypass this issue is to rename already existing in map nodes and
+			// set them new unique name.
+			if ( prevFakeMarker && prevFakeMarker.start && prevFakeMarker.end ) {
+				acc[ this._getUniqueMarkerName( fakeMarker.name ) ] = acc[ fakeMarker.name ];
+				prevFakeMarker = null;
+			}
 
-				// [ foo <fake-marker>aaa</fake-marker> test ]
-				//                    ^
-				// Handle case when marker is between start and end of selection.
-				const startPosition = writer.createPositionAt( startElement, 'before' );
-				writer.remove( startElement );
+			acc[ fakeMarker.name ] = {
+				...prevFakeMarker!,
+				[ fakeMarker.type ]: position
+			};
 
-				const endPosition = writer.createPositionAt( endElement, 'before' );
-				writer.remove( endElement );
+			if ( fakeMarker.markerElement ) {
+				writer.remove( fakeMarker.markerElement );
+			}
 
-				acc[ markerName ] = new Range( startPosition, endPosition );
+			return acc;
+		}, {} );
 
-				return acc;
-			}, {} );
+		return mapValues(
```

Comment:
Do I understand correctly that we need to do it at the end of all processing, because the `rootElement` changes and `'end'` position may vary?

If so, let's add a comment about it because it is not so obvious on first glance to a newcomer.

Reply:
Added